src/infra/core/src/ingest/push_ingest_service_impl.rs

```diff
@@ -0,0 +1,368 @@
+            time_source,
+        }
+    }
+
+    async fn do_ingest(
+        &self,
+        dataset_ref: &DatasetRef,
+        url: url::Url,
+        media_type: Option<MediaType>,
+        listener: Arc<dyn PushIngestListener>,
+    ) -> Result<PushIngestResult, PushIngestError> {
+        let dataset_handle = self.dataset_repo.resolve_dataset_ref(&dataset_ref).await?;
+
+        self.dataset_action_authorizer
+            .check_action_allowed(&dataset_handle, auth::DatasetAction::Write)
+            .await?;
+
+        let dataset = self
+            .dataset_repo
+            .get_dataset(&dataset_handle.as_local_ref())
+            .await?;
+
+        // TODO: Temporarily relying on SetPollingSource event
+        let Some(polling_source) = dataset
+            .as_metadata_chain()
+            .last_of_type::<SetPollingSource>()
+            .await
+            .int_err()?
+            .map(|(_, b)| b.event)
+        else {
+            let err = PushIngestError::SourceNotFound(PushSourceNotFoundError);
+            listener.begin();
+            listener.error(&err);
+            return Err(err);
+        };
+
+        let operation_id = ingest_common::next_operation_id();
+        let operation_dir = self.run_info_dir.join(format!("ingest-{}", operation_id));
+        std::fs::create_dir_all(&operation_dir).int_err()?;
+
+        let ctx: SessionContext =
+            ingest_common::new_session_context(self.object_store_registry.clone());
+        let data_writer = DataWriterDataFusion::builder(dataset.clone(), ctx.clone())
+            .with_metadata_state_scanned()
+            .await?
+            .build()
+            .await?;
+
+        let args = PushIngestArgs {
+            operation_id,
+            operation_dir,
+            system_time: self.time_source.now(),
+            url,
+            media_type,
+            listener,
+            ctx,
+            data_writer,
+            polling_source,
+        };
+
+        let listener = args.listener.clone();
+        listener.begin();
+
+        match self.do_ingest_inner(args).await {
+            Ok(res) => {
+                tracing::info!(result = ?res, "Ingest iteration successful");
+                listener.success(&res);
+                Ok(res)
+            }
+            Err(err) => {
+                tracing::error!(error = ?err, "Ingest iteration failed");
+                listener.error(&err);
+                Err(err)
+            }
+        }
+    }
+
+    #[tracing::instrument(
+        level = "info",
+        skip_all,
+        fields(
+            operation_id = %args.operation_id,
+        )
+    )]
+    async fn do_ingest_inner(
+        &self,
+        mut args: PushIngestArgs,
+    ) -> Result<PushIngestResult, PushIngestError> {
+        args.listener
+            .on_stage_progress(PushIngestStage::CheckSource, 0, TotalSteps::Exact(1));
+        args.listener
+            .on_stage_progress(PushIngestStage::Fetch, 0, TotalSteps::Exact(1));
```

Comment:
Exactly what I was talking about - if we don't need it , why we still keep it.

Reply:
As per other comment - will remove these stages